src/components/DetailsOverlay/index.tsx

```diff
@@ -0,0 +1,77 @@
+import styled from "@emotion/styled";
+import Filter from "components/Filter";
+import { QueryParams } from "constants/queryParams";
+import { size } from "constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import ButtonRow from "./ButtonRow";
+import SearchRangeInput from "./SearchRangeInput";
+import {
+  CaseSensitiveToggle,
+  ExpandableRowsToggle,
+  FilterLogicToggle,
+  LogV2Toggle,
+  PrettyPrintToggle,
+  WrapToggle,
+} from "./Toggle";
+
+interface DetailsOverlayProps {
+  ["data-cy"]?: string;
+}
+
+const DetailsOverlay: React.FC<DetailsOverlayProps> = ({
+  "data-cy": dataCy,
+}) => {
+  const [filters, setFilters] = useQueryParam<string[]>(
+    QueryParams.Filters,
+    []
+  );
+
+  const deleteFilter = (filterName: string) => {
+    const newFilters = filters.filter((f) => f !== filterName);
+    setFilters(newFilters);
+  };
+
+  return (
+    <Card data-cy={dataCy}>
+      <FilterOptions>
+        <FilterContainer>
+          <SearchRangeInput />
+          <WrapToggle />
+          <CaseSensitiveToggle />
+          <LogV2Toggle />
+        </FilterContainer>
+        <FilterContainer>
+          <FilterLogicToggle />
+          <ExpandableRowsToggle />
+          <PrettyPrintToggle />
+          <ButtonRow />
+        </FilterContainer>
+      </FilterOptions>
+
+      {filters.length !== 0 && <Divider />}
+      {filters.map((filter) => (
+        <Filter key={filter} deleteFilter={deleteFilter} filterText={filter} />
+      ))}
```

Comment:
Allot of filters overlap the page and cause us to overflow.
Limiting the height of the container may help/ introducing some scroll.
<img width="1509" alt="image" src="https://user-images.githubusercontent.com/4605522/190255722-bc45103b-eb2f-428f-b926-483e00896226.png">